feat/v1.5.1 (#217)
smartmind-team/thanosql-docs: docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md

Reviewer: Let's split the API example into two methods. Mixing them all into one example makes it seem even more confusing.

```
## __`POST` /query__

### Query without Templates

### Query with Templates
```

    import json

    api_token = "Issued_API_TOKEN"
    api_url = "https://{your-engine-url}/api/v1/query/"
    query_type = "SQL query type" #psql or thanosql

    header = {
        "Authorization": f"Bearer {api_token}"
    }

    # For query without templates (pick 1 between this and below):
    query = "Query to request"

    data = {
        'query_string': query, 'query_type': query_type
    }

    # For query with templates (pick 1 between this and above):
    template_name = "Name of template to use"
    parameters = "Mapping of parameters to fill in the template"

    data = {
        'query_type': query_type,

>         'template_name': template_name,  # or 'template_id'

Reviewer 2: I think we need to be more verbose with our description regarding template_id

    import json

    api_token = "Issued_API_TOKEN"
    api_url = "https://{your-engine-url}/api/v1/query/"

>     query_type = "SQL query type" #psql or thanosql

Reviewer: ```suggestion
    query_type = "SQL query type" # psql or thanosql
```
ko도 바꿔주세요

    header = {
        "Authorization": f"Bearer {api_token}"
    }

    # For query without templates (pick 1 between this and below):
    query = "Query to request"

    data = {
        'query_string': query, 'query_type': query_type
    }

    # For query with templates (pick 1 between this and above):
    template_name = "Name of template to use"     # we don't need this line if we query by template_id
    template_id = "ID number of template to use"  # we don't need this line if we query by template_name

>     parameters = "Mapping of parameters to fill in the template"

Reviewer: ```suggestion
    parameters = "Mapping of parameters to fill in the template" # Dictionary
```
string이 아니라서 명시해줘야 할 것 같습니다. (ko도 마찬가지)

    api_token = "Issued_API_TOKEN"
    base_url = "https://{your-engine-url}/api/v1/query/log"
    search = "Search keyword(s)"
    offset = {Offset}
    limit = {Limit}

    api_url = f"{base_url}?search={search}&offset={offset}&limit={limit}"

    header = {
        "Authorization": f"Bearer {api_token}"
    }

    r = requests.get(api_url, headers=header):
    r.raise_for_status()
    r.json()
    ```

=== "cURL"

    ```shell
      curl -X 'GET' \
      'https://{your-engine-url}/api/v1/query/log/?search={search}&offset={offset}&limit={limit}' \
      -H 'accept: application/json' \
      -H 'Authorization: Bearer Issued_API_TOKEN'
    ```

### __Parameters__

> - `search`: Word(s) that the returned query logs should contain.

Reviewer 2: ```suggestion
- `search`: Word(s) that the query in the query log should contain.
```
We are searching for words in the query part of the query log so change the description to something in the same grain as above

    ```sql
    GET THANOSQL DATASET diet_data
    OPTIONS (overwrite=True)
    ```

## **`POST` /query/template**

Creates a new query template with a certain name and query string.

=== "Python"

    ```python
    import requests
    import json

    api_token = "Issued_API_TOKEN"
    api_url = "https://{your-engine-url}/api/v1/query/template"
    name = "Name of query template"
    query = "Query template string (in psql or thanosql)"

    header = {
        "Authorization": f"Bearer {api_token}"
    }

    data = {
        'name': name, 'query': query
    }

    r = requests.post(api_url, data=json.dumps(data), headers=header)
    r.raise_for_status()
    r.json()
    ```

=== "cURL"

    ```shell
    curl -X 'POST' \
      'https://{your-engine-url}/api/v1/query/template' \
      -H 'accept: application/json' \
      -H 'Authorization: Bearer Issued_API_TOKEN' \
      -H 'Content-Type: application/json' \
      -d '{"name": name, "query": query}'
    ```

## **`GET` /query/template/{template_name}**

> Retrieves the query template with a certain name.

Reviewer 2: ```suggestion
Retrieves the query template with the given name.
```

=== "Python"

    ```python
    import requests
    import json

    api_token = "Issued_API_TOKEN"
    base_url = "https://{your-engine-url}/api/v1/query/template"
    template_name = "Name of query template"

    api_url = f"{base_url}/{template_name}"

    header = {
        "Authorization": f"Bearer {api_token}"
    }

    r = requests.get(api_url, headers=header)
    r.raise_for_status()
    r.json()
    ```

=== "cURL"

    ```shell
    curl -X 'GET' \
      'https://{your-engine-url}/api/v1/query/template/{template_name}' \
      -H 'accept: application/json' \
      -H 'Authorization: Bearer Issued_API_TOKEN'
    ```

## **`PUT` /query/template/{template_name}**

> Updates a query template with a certain name.

Reviewer 2: ```suggestion
Updates the query template with the given name.
```

=== "Python"

    ```python
    import requests
    import json

    api_token = "Issued_API_TOKEN"
    base_url = "https://{your-engine-url}/api/v1/query/template"
    template_name = "Name of query template"
    name = "Updated name of query template"
    query = "Updated query template string (in psql or thanosql)"

    api_url = f"{base_url}/{template_name}"

    header = {
        "Authorization": f"Bearer {api_token}"
    }

    data = {
        'name': name, 'query': query
    }

    r = requests.put(api_url, data=json.dumps(data), headers=header)
    r.raise_for_status()
    r.json()
    ```

=== "cURL"

    ```shell
    curl -X 'PUT' \
      'https://{your-engine-url}/api/v1/query/template/{template_name}' \
      -H 'accept: application/json' \
      -H 'Authorization: Bearer Issued_API_TOKEN' \
      -H 'Content-Type: application/json' \
      -d '{"name": name, "query": query}'
    ```

!!! Note
    If you only want to update one field, leave the other field out of the data JSON. For example, if you only want to update the name of the query template, use `{"query": query}` as the data body.

## **`DELETE` /query/template/{template_name}**

> Deletes a query template with a certain name.

Reviewer 2: ```suggestion
Deletes the query template with the given name.
```
